src/parser.ml

```diff
@@ -1706,9 +1741,10 @@ and
 
 end
 
-let parse_decls lang dataModel enforceAnnotations ?inGhostHeader =
+let parse_decls (lang, dialect_opt) dataModel enforceAnnotations ?inGhostHeader =

```

Comment:
Is it really necessary to pass a dialect argument to the parser? Are there any programs that are valid in two languages but have a different meaning? It is okay if the VeriFast parser accepts more programs than the compiler. Also, note that we can consider some words as keywords in one language, and regular identifiers in another language.

Reply:
Removed the optional dialect